task/buildah-rhtap/0.1/buildah-rhtap.yaml

```diff
@@ -0,0 +1,197 @@
+
+      # Set task results
+      buildah images --format '{{ .Name }}:{{ .Tag }}@{{ .Digest }}' | grep -v $IMAGE > $(results.BASE_IMAGES_DIGESTS.path)
+      cat /tmp/files/image-digest | tee $(results.IMAGE_DIGEST.path)
+      echo -n "$IMAGE" | tee $(results.IMAGE_URL.path)
+
+      # Save the image so it can be used in the generate-sbom step
+      buildah push "$IMAGE" oci:/tmp/files/image
+    securityContext:
+      capabilities:
+        add:
+          - SETFCAP
+    volumeMounts:
+    - mountPath: /var/lib/containers
+      name: varlibcontainers
+    - mountPath: /tmp/files
+      name: tmpfiles
+    workingDir: $(workspaces.source.path)
+
+  - name: generate-sboms
+    image: quay.io/redhat-appstudio/syft:v0.98.0@sha256:4d3856e6a2622700b9a9d5d74d9aaf5d8a55671653f80bf6c636677658680ede
+    script: |
+      syft dir:$(workspaces.source.path)/source --output cyclonedx-json@1.4=/tmp/files/sbom-source.json
+      syft oci-dir:/tmp/files/image --output cyclonedx-json@1.4=/tmp/files/sbom-image.json
+    volumeMounts:
+    - mountPath: /var/lib/containers
+      name: varlibcontainers
+    - mountPath: /tmp/files
+      name: tmpfiles
+
+  - name: merge-sboms
+    image: registry.access.redhat.com/ubi8/python-311@sha256:8ded4b6d8087706b6819ddda5d31f22b80e5aa4efa772e94d750699ccfbf98eb
+    env:
+      - name: SBOM_BLOB_URL
+        value: $(results.SBOM_BLOB_URL.path)
+    script: |
+      #!/bin/python3
+      import hashlib
+      import json
+      import os
+
+      ### load SBOMs ###
+
+      with open("./sbom-image.json") as f:
+        image_sbom = json.load(f)
+
+      with open("./sbom-source.json") as f:
+        source_sbom = json.load(f)
+
+
+      ### attempt to deduplicate components ###
+
+      component_list = image_sbom.get("components", [])
+      existing_purls = list(filter(lambda c: "purl" in c, component_list))
+
+      for component in source_sbom.get("components", []):
+        if "purl" in component:
+          if component["purl"] not in existing_purls:
+            component_list.append(component)
+            existing_purls.append(component["purl"])
+        else:
+          # We won't try to dedup components that lack a purl.
+          # This should only happen with operating-system type components,
+          # which are only reported in the image SBOM.
+          component_list.append(component)
+
+      component_list.sort(key=lambda c: c["type"] + c["name"])
+      image_sbom["components"] = component_list
+
+
+      ### write the CycloneDX unified SBOM ###
+
+      with open("./sbom-cyclonedx.json", "w") as f:
+        json.dump(image_sbom, f, indent=4)
+
+
+      ### write the SBOM digest result ###
+
+      with open("./sbom-cyclonedx.json", "rb") as f:
+        sbom_digest = hashlib.file_digest(f, "sha256").hexdigest()
+
+      image_repo = os.getenv("IMAGE").split(":")[0]
+      sbom_blob_url = f"{image_repo}@sha256:{sbom_digest}"
+
+      print(image_repo)
+      print(sbom_blob_url)
```

Comment:
Are the prints intentional?

Reply:
Nope, they were accidentally commited in some rebase :face_exhaling:

Reply:
This commit was entirely dropped.